llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -135,6 +138,12 @@ static cl::opt<unsigned> FPAssociationUpperLimit(
         "Set upper limit for the number of transformations performed "
         "during a single round of hoisting the reassociated expressions."));
 
+cl::opt<unsigned> IntAssociationUpperLimit(
+    "licm-max-num-int-reassociations", cl::init(5U), cl::Hidden,
+    cl::desc(
+        "Set upper limit for the number of transformations performed "
+        "during a single round of hoisting the reassociated expressions."));
+
```

Comment:
Is the type relevant to the limit?  I'm not against the change just wondering if this is a compile time reduction flag whether it's worth just renaming the current one.